pennylane/templates/layers/particle_conserving_u2.py

```diff
@@ -246,3 +255,67 @@ def shape(n_layers, n_wires):
                 f"The number of qubits must be greater than one; got 'n_wires' = {n_wires}"
             )
         return n_layers, 2 * n_wires - 1
+
+
+def _particle_conserving_u2_resources(num_wires: int, n_layers: int):
+    # number of pairs of even-indexed of wires
+    num_nm_wires = (
+        math.floor(num_wires / 2) if num_wires % 2 == 0 else math.ceil((num_wires - 1) / 2)
+    )
+    # number of odd-indexed pairs of wires
+    num_nm_wires += math.floor((num_wires - 1) / 2)
```

Comment:
Could also simplify the overall count if you like (but could also keep the two steps to relate them to the even and odd parts, either way seems good).

```suggestion
    num_nm_wires = num_wires - 1
```